Mathlib/Data/Matrix/Rank.lean

```diff
@@ -119,6 +119,18 @@ lemma rank_mul_eq_right_of_isUnit_det [Fintype m] [DecidableEq m]
   have hAB : mulVecLin (A * B) = (LinearEquiv.ofIsUnitDet hA).comp (mulVecLin B) := by ext; simp
   rw [rank, rank, hAB, LinearMap.range_comp, LinearEquiv.finrank_map_eq]
 
+variable {m n R : Type*} [Fintype n] [CommRing R] [DecidableEq n]
```

Comment:
It's not necessary to shadow all of these variables since most (all?) are in context already. Any variables that are still needed should be put directly in the theorem statement.

Reply:
OK. Will do. I ended up doing this in the unitary part because some things broke when I tried to include this...